fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/directory/AgilityContext.java

```diff
@@ -170,6 +170,8 @@ default <T> T getPropertyValue(@Nonnull RecordLayerPropertyKey<T> propertyKey) {
         return getCallerContext().getPropertyStorage().getPropertyValue(propertyKey);
     }
 
+    void setPreCommit(Function<FDBRecordContext, CompletableFuture<Void>> preCommit);
```

Comment:
Would you add a JavaDoc comment explaining that the `preCommit` callback is expected to use the argument context but should never call other agility context functions as it may cause a deadlock?